uk.co.compucorp.civicrm.hrleaveandabsences/gulpfile.js

```diff
@@ -104,24 +103,35 @@ var test = (function () {
     /**
      * Runs the tests for a specific source file
      *
-     * Looks for a test file (*_test.js) in `test/`, using the same path
+     * Looks for a test file (*_test.js or *.spec.js) in `test/`, using the same path
      * of the source file in `src/leave-absences/`
      *   i.e. src/leave-absences/models/model.js -> test/models/model_test.js
+     *   or   src/leave-absences/models/model.js -> test/models/model.spec.js
+     *   @TODO test files *_test.js should be renamed to *.spec.js
      *
      * @throw {Error}
      */
     for: function (srcFile) {
       var srcFileNoExt = path.basename(srcFile, path.extname(srcFile));
-      var testFile = srcFile
-        .replace('src/leave-absences/', 'test/')
-        .replace(srcFileNoExt + '.js', srcFileNoExt + '_test.js');
+      var testFileFound = false;
+      var testFileSuffixes = [ '_test.js', '.spec.js' ];
+
+      testFileSuffixes.forEach(function (testFileSuffix) {
+        var testFile = srcFile
+          .replace('src/leave-absences/', 'test/')
+          .replace(srcFileNoExt + '.js', srcFileNoExt + testFileSuffix);
+
+        try {
+          var stats = fs.statSync(testFile);
+
+          stats.isFile() && this.single(testFile);
 
-      try {
-        var stats = fs.statSync(testFile);
+          testFileFound = true;
+        } catch (ex) {}
+      });
 
-        stats.isFile() && this.single(testFile);
-      } catch (ex) {
-        throw ex;
+      if (!testFileFound) {
+        throw new Error('Test file not found');
```

Comment:
this is not really necessary, the `fs.statSync` in the `try` block already throws an exception, which is then used to run all the tests (as in, if a specific test file is not found, run all of them)